README.md

```diff
@@ -182,6 +182,100 @@ curl http://127.0.0.1:8080/gotify_webhook -d '
 ]}
 '
 ```
+### Bridge Message Templating
+The bridge now supports user-defined templating for all inbound messages. The user-defined templating can be used for the title and/or message. Also, user-defined templating overrides the default title and message annotations. All keys and values in the JSON from alertmanager can be used in the user-defined template. Any failures in the templates will result in the bridge defaulting back to default alerting.
+
+#### Usage Notes:
+- For Docker, you must bind your volume to your host to add user-defined templating.
+- The default directory for all templates is the root of the bridge in the folder called `templates`.
+- User-defined templating allows matching and linking using the "define" name of the template [Go Templating](https://golang.google.cn/pkg/text/template/).
+- The Gotify software token is used for matching a message template.
+- The Gotify software token with `title=` at the beginning is used for matching a title template.
+- All file names must be unique but can be any name or subfolder naming you choose.
```

Comment:
My favorite part about seeing the power of Grafana was the complete customization they offer with Go Templating. My thoughts with the bridge were to allow the same full unrestricted customization for the bridge's templates.

For example, I may have a template for logging that I want to use for several different "applications" in Gotify. I can link this by using the defined variable quickly and efficiently without extra effort. The same could be said for title templates and broader message templates (networking, logging, etc.).

I see how the file structure could help users unfamiliar with Go Templating, but I suspect users unfamiliar with Go Templating would use the bridge's default messaging. My guess is that users using the bridge have Grafana installed and are familiar with the Go Templating in Grafana, which would allow them to use this functionality without any confusion. With a defined folder structure, the template would still need a defined variable (name) to match during the ExecuteTemplate function unless you have seen this handled differently.

Reply:
Ah, yes - my thought is that since the go templating system [will default to the file name being used as the name of the template](https://pkg.go.dev/text/template#Template.ParseFiles) (as though it were `define`'ed in the template), we could rely on the file names being the template name unless decided otherwise by the user (placing a `define` inside). This would help a lot during template application since we can then use `template.Lookup("title")` during processing to get the title template. It also opens the door to the user to be as powerful as they'd like by defining many sub-templates (a feature I hadn't considered until I saw it in this PR) in various other files.

&nbsp;

> I see how the file structure could help users unfamiliar with Go Templating, but I suspect users unfamiliar with Go Templating would use the bridge's default messaging. My guess is that users using the bridge have Grafana installed and are familiar with the Go Templating in Grafana, which would allow them to use this functionality without any confusion

Interestingly enough, I don't use Grafana templates at all! I only use the [Prometheus alertmanager](https://prometheus.io/docs/alerting/latest/alertmanager/), which was the genesis of this particular project 😄 It wasn't until prior PRs discussed the Grafana templating capabilities that I even thought Grafana would be used as a source of alerts for the bridge.

&nbsp;
Hopefully the [longer comment](https://github.com/DRuggeri/alertmanager_gotify_bridge/pull/29#pullrequestreview-1304850857) I left explains the thought process. By loading as much up front as possible, supporting template overrides and customization via directory trees, and using shortcuts available thanks to template lookups by name, I think this will give the flexibility you describe while also being lean during message processing.

Reply:
I did not realize Go Templating defaulted to the file name. That is good to know.

Good point on you not using Grafana templating. Since it is limited I am sure others have not even tried it yet. It has great potential if they allow it to alert with their alert engine when Prometheus alerts trigger. The limitation of Grafana is what allowed me to come across this bridge. I still believe people with an understanding of structuring single-file Go Templates (especially the functions), would be able to pick up on defining templates fairly quickly.

My original PR could load everything up front as well. I just had it loading during the handleCall for dynamic user template changes. The load would have no dependencies on what users call tokens are being sent. All user templates would load and if a token was sent during an alert, a matching token would utilize that template.

One issue I see with the folder `token` structure is special OS limitations on naming. Linux allows you to end a folder with a period, but Windows does not allow a folder to end with a period. The period would be stripped from any instance of the bridge running on Windows. Tokens are randomly generated with no ability to edit/change and could lead to issues with people implementing templating with existing applications. I have some applications in my Gotify that end with a period. We could try and add code to detect if a folder has 99% of the values, but we would be adding additional overhead to the bridge.

Few things from your high-level thought process:
1. The function example you listed would need some editing to handle sub-directories. Any root (common) templates would need to `Must` with the return *ut.Template as a variable, so that variable can pass to the ParseGlob for any sub-directory. Without this, you would get "not defined" errors when referencing templates in the sub-directories.
2. Restricting the files to only tmpl would restrict users from using other popular file formats, that only syntax color in certain code editors (ex: VS Code).